kamon-core/src/main/scala/kamon/util/executors/ExecutorMetricRecorder.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * =========================================================================================
+ * Copyright © 2013-2015 the kamon project <http://kamon.io/>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ * =========================================================================================
+ */
+
+package kamon.util.executors
+
+import akka.dispatch.ForkJoinExecutorConfigurator.AkkaForkJoinPool
+import kamon.metric.{ EntityRecorderFactory, GenericEntityRecorder }
+import kamon.metric.instrument.{ Gauge, MinMaxCounter, DifferentialValueCollector, InstrumentFactory }
+import java.util.concurrent.{ ForkJoinPool ⇒ JavaForkJoinPool, ThreadPoolExecutor }
+import kamon.util.executors.ForkJoinPools.ForkJoinMetrics
+
+import scala.concurrent.forkjoin.ForkJoinPool
+
+object ForkJoinPools {
+  trait ForkJoinMetrics[T] {
+    def getParallelism(fjp: T): Long
+    def getPoolSize(fjp: T): Long
+    def getActiveThreadCount(fjp: T): Long
+    def getRunningThreadCount(fjp: T): Long
+    def getQueuedTaskCount(fjp: T): Long
+  }
+
+  implicit object ScalaForkJoin extends ForkJoinMetrics[ForkJoinPool] {
+    def getParallelism(fjp: ForkJoinPool) = fjp.getParallelism
+    def getPoolSize(fjp: ForkJoinPool) = fjp.getPoolSize.toLong
+    def getRunningThreadCount(fjp: ForkJoinPool) = fjp.getActiveThreadCount.toLong
+    def getActiveThreadCount(fjp: ForkJoinPool) = fjp.getRunningThreadCount.toLong
+    def getQueuedTaskCount(fjp: ForkJoinPool) = fjp.getQueuedTaskCount
+
+  }
+
+  implicit object JavaForkJoin extends ForkJoinMetrics[JavaForkJoinPool] {
+    def getParallelism(fjp: JavaForkJoinPool) = fjp.getParallelism
+    def getPoolSize(fjp: JavaForkJoinPool) = fjp.getPoolSize.toLong
+    def getRunningThreadCount(fjp: JavaForkJoinPool) = fjp.getActiveThreadCount.toLong
+    def getActiveThreadCount(fjp: JavaForkJoinPool) = fjp.getRunningThreadCount.toLong
+    def getQueuedTaskCount(fjp: JavaForkJoinPool) = fjp.getQueuedTaskCount
+
+  }
+
+  implicit object AkkaForkJoin extends ForkJoinMetrics[AkkaForkJoinPool] {
```

Comment:
you are right, indeed i've replaced in DispatcherInstrumentation the type of the Dispatcher  to ForkJoinPool instead of AkkaForkJoinPool